src/components/user/uc/users.uc.js

```diff
@@ -0,0 +1,67 @@
+const { ObjectId } = require('mongoose').Types;
+const { GeneralError, NotFound } = require('../../../errors');
+const { userRepo, accountRepo, trashbinRepo } = require('../repo/index');
+const restrictToSameSchool = require('../../../utils/restrictToSameSchool');
+const restrictToRole = require('../../../utils/restrictToRole');
+
+const getUserData = async (id, app) => {
+	const data = {};
+
+	const user = await userRepo.getUser(id, app);
+	if (!(user && user._id && !user.deletedAt)) {
+		throw new NotFound(`User ${id} not found`);
+	}
+	data.user = user;
+
+	const account = await accountRepo.getUserAccount(id, app);
+	if (account) {
+		data.account = account;
+	}
+	return data;
+};
+
+const deleteUserData = async (id, app) => {
+	await accountRepo.deleteUserAccount(id, app);
+};
+
+const createUserTrashbin = async (id, data) => {
+	const trashBin = await trashbinRepo.createUserTrashbin(id, data);
+	if (!(trashBin && trashBin._id)) {
+		throw new GeneralError(`Unable to initiate trashBin`);
+	}
+	return trashBin;
+};
+
+const replaceUserWithTombstone = async (id, app) => {
+	const uid = ObjectId();
+	await userRepo.replaceUserWithTombstone(
+		id,
+		{
+			firstName: 'DELETED',
+			lastName: 'USER',
+			email: `${uid}@deleted`,
+			deletedAt: new Date(),
+		},
+		app
+	);
+	return { success: true };
+};
+
+const deleteUserUC = async (id, roleName, { account, app }) => {
+	await restrictToSameSchool(id, account, app);
```

Comment:
try catch around all stuff inside should set, because deleteUserUC is export use cases and should valid to test and catch errors in chain that is handled. If you want to split up in different errors please use more then one try catch

Reply:
createUserTrashbin, replaceUserWithTombstone, deleteUserData give the error from the repo layer without interpretation to the service layer. With try catch you not lose the error it is only interpreted. 
At the moment we not have the application errors that we need at this position but one error with context must throw.

catch (err) {
   throw new Unprocessable('Deletion of...', err);
}

The repo layer need in the must situation no error handling. But UCs should throw application errors and the service layer the http errors.